ConceptualAPI.java

```diff
@@ -0,0 +1,17 @@
+/* Step 1: Identify the user
+ *    the user is the component initializing and coordinating the data
+ * 
+ * Step 2: What the user needs to do
+ *      initialize the job
+ *      tell other component to compute data
+ *      organize output
+ *   
+ * Step 3: Build a prototype
+ * 
+ */
+public interface ConceptualAPI {
+
+    JobInitializeResponse initializeJob(InitializeRequest request);

```

Comment:
These three can actually all be combined into one - you'd never have a situation where a user shouldn't immediately call all 3 right after the other (compared to something like setting a delimiter, which a user might or might not end up doing, or read/write calls that are separated by logic that depends on the results of read).